test/suites/basic.sh

```diff
@@ -672,6 +672,27 @@ test_basic_usage() {
   lxc profile delete foo
   lxc delete -f c1
 
+  # Test assigning a profile through a YAML file to an instance.
+  poolName=$(lxc profile device get default root pool)
+  lxc profile create foo < <(cat <<EOF
+config:
+  limits.cpu: 2
+  limits.memory: 1024MB
+description: Test profile
+devices:
+  root:
+    path: /
+    pool: ${poolName}
+    type: disk
+EOF
+)
+  lxc init testimage c1
+  lxc profile assign c1 foo
+  lxc config show c1 --expanded | grep -q "limits.cpu: \"2\""
```

Comment:
Please can you use this style of explicit check https://github.com/canonical/lxd/commit/3d0ae50dd35539acdd1248f4ef50e746e7ff977e this is simpler and easier to debug when it fails in the tests as we can see the "wrong" value.

Reply:
fixed